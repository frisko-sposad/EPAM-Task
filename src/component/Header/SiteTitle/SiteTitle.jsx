import React from 'react';
import './SiteTitle.css';

const SiteTitle = () => {
  return <h1 className="site-title">NetflixRoulette</h1>;
};

export default SiteTitle;
