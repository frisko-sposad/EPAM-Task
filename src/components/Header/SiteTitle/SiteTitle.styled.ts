import styled from 'styled-components';

const SiteTitleH1 = styled.h1`
  margin: 0 auto;
  max-width: 1060px;
  padding: 1rem 3rem;
`;

export default SiteTitleH1;
