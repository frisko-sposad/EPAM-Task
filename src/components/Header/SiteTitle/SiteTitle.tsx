import React, { FC } from 'react';
import './SiteTitle.css';

const SiteTitle: FC = () => <h1 className="site-title">NetflixRoulette</h1>;

export default SiteTitle;
