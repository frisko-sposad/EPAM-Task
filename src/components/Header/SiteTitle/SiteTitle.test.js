import React from 'react';
import { shallow } from 'enzyme';
import SiteTitle from './SiteTitle';

const setUp = (props) => shallow(<SiteTitle {...props} />);

describe('render SiteTitle', () => {
  let component;
  beforeEach(() => {
    component = setUp();
  });

  it('render header title typeof = string', () => {
    const wrapper = component.find('h1');
    const result = wrapper.text();
    expect(typeof result).toBe('string');
  });
  it('render header title', () => {
    const wrapper = component.find('h1');
    expect(wrapper.length).toBe(1);
  });
  
  describe('Snapshot SiteTitle', () => {
    it('Snapshot SiteTitle', () => {
      expect(component).toMatchSnapshot();
    });
  });
});
