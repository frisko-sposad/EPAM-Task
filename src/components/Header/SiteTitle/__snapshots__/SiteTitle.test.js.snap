// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render SiteTitle Snapshot SiteTitle Snapshot SiteTitle 1`] = `
<h1
  className="site-title"
>
  NetflixRoulette
</h1>
`;
